api/src/main/java/io/opentelemetry/trace/Sampler.java

```diff
@@ -58,4 +59,30 @@ boolean shouldSample(
    * @since 0.1.0
    */
   String getDescription();
+
+  /**
+   * Sampling decision returned by {@link Sampler#shouldSample(SpanContext, Boolean, TraceId,
+   * SpanId, String, List)}.
+   *
+   * @since 0.1.0
+   */
+  interface Decision {
+
+    /**
+     * Return sampling decision whether span should be sampled or not.
+     *
+     * @return sampling decision.
+     * @since 0.1.0
+     */
+    boolean isSampled();
+
+    /**
+     * Return tags which will be attached to the span.
```

Comment:
I think this is a good idea. Some attributes needs to be added to the `tracestate`. We can defer this need or enable it.

Reply:
https://github.com/open-telemetry/opentelemetry-java/issues/357